tests/integration/hooks.bats

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bats
+
+load helpers
+
+# CR = CreateRuntime
+# CC = CreataContainer
+HOOKLIBCR=librunc-hooks-create-runtime.so
+HOOKLIBCC=librunc-hooks-create-container.so
+LIBPATH="$DEBIAN_BUNDLE/rootfs/lib/"
+
+function setup() {
+	umount $LIBPATH/$HOOKLIBCR.1.0.0 &> /dev/null || true
+	umount $LIBPATH/$HOOKLIBCC.1.0.0 &> /dev/null || true
+
+	teardown_debian
+	setup_debian
+}
+
+function teardown() {
+	umount $LIBPATH/$HOOKLIBCR.1.0.0 &> /dev/null || true
+	umount $LIBPATH/$HOOKLIBCC.1.0.0 &> /dev/null || true
+
+	rm -f $HOOKLIBCR.1.0.0 $HOOKLIBCC.1.0.0
+	teardown_debian
+}
+
+@test "runc run (hooks library tests)" {
+	requires root
+	requires no_systemd
+
+	# setup some dummy libs
+	gcc -shared -Wl,-soname,librunc-hooks-create-runtime.so.1 -o "$HOOKLIBCR.1.0.0"
+	gcc -shared -Wl,-soname,librunc-hooks-create-container.so.1 -o "$HOOKLIBCC.1.0.0"
```

Comment:
As happy as I am to have `umoci` as a build dependency for runc, as far as I can tell the only reason you need debian is because you want to run `ldconfig` as a way to check if certain files exist in the container. Is it possible to just create files with special contents and check that the contents are correct -- that way you wouldn't need a debian image for testing? Or is there a particular reason you want to test `ldconfig` specifically?

Reply:
To avoid blocking this PR for too long, I'd be happy to take it as-is and we can later change all of the tests to use Debian rather than busybox (this would also avoid having two methods of downloading and unpacking image as well). I'm not sure if that will be a painful change to make, so let's punt on it for now.

I don't want to over-simplify the use-cases you folks have (even though `ldconfig` is ultimately "just writing files") so I'd prefer to move the rest of the test suite rather than over-simplifying this particular test.

Reply:
I've got some time set aside to refactor the Hooks after this, I can add some more to refactor the tests :) !